src/types.ts

```diff
@@ -16,9 +16,10 @@ export interface StorageMeta {
 
 export type TransactionOptions = Record<string, any>;
 
-export interface Driver {
+export interface Driver<T = undefined> {
```

Comment:
For future compatibiliy, drivers might use first generic for Options type so it would be better something like this:

```suggestion
export interface Driver<OptionsT = any, InstanceT = any> {
```

Reply:
Improving `Driver` type introduces a bit more type errors and I tried to fix them.